packages/block-editor/src/store/selectors.js

```diff
@@ -1328,6 +1328,29 @@ export const hasInserterItems = createSelector(
 	],
 );
 
+/**
+ * Determines whether there is only one item that may be inserted.
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {boolean} True if there is one item available, false if zero or more than one.
+ */
+export const hasOneAllowedItem = createSelector(
```

Comment:
Was there a reason to cache this selector? It neither returns a new reference upon each new call, nor is it expensive (its dependency, `getBlockListSettings`, is a simple object access).

Reply:
no there was no reason other than me not having enough experience with wp data and not paying attention to what createselector does :)